File: JavaWebApi/src/main/java/com/laixhe/result/Result.java
package com.laixhe.result;

import lombok.Data;

import java.io.Serializable;

import com.laixhe.exception.BusinessException;

/**
 * @author laixhe
 */
@Data
public class Result<T> implements Serializable {

    private int code;
    private String msg;
    private T data;

    public static <T> Result<T> success() {
        return success(null);
    }

    public static <T> Result<T> success(T data) {
        Result<T> result = new Result<>();
        result.setCode(ResultCode.Success.getCode());
        result.setMsg(ResultCode.Success.getMsg());
        result.setData(data);
        return result;
    }

    public static <T> Result<T> error(String msg) {
        Result<T> result = new Result<>();
        result.setCode(ResultCode.Service.getCode());
        result.setMsg(msg);
        return result;
    }

    // 自定义异常返回的结果
    public static <T> Result<T> businessErr(BusinessException e) {
        Result<T> result = new Result<>();
        result.setCode(e.getCode().getCode());
        result.setMsg(e.getMsg().isBlank() ? e.getCode().getMsg() : e.getMsg());
        result.setData(null);
        return result;
    }

    // 其他异常处理方法返回的结果
    public static <T> Result<T> otherErr(ResultCode resultCode, String msg) {
        Result<T> result = new Result<>();
        result.setCode(resultCode.getCode());
        result.setMsg(msg.isBlank() ? resultCode.getMsg() : msg);
        result.setData(null);
        return result;
    }


}

File: PyWebApi/app/result/result.py
from typing import Any
from pydantic import BaseModel, Field


class Result(BaseModel):
    """ http统一响应 """
    code: int = Field(default=0)  # 响应码
    msg: str = Field(default='')  # 响应信息
    data: Any | None  # 具体数据


def success(resp: Any) -> Result:
    """ 成功响应 """
    return Result(data=resp)


def fail(msg: str, code: int = 1) -> Result:
    """ 响应失败 """
    return Result(
        code=code,
        msg=msg)
